config/backend-config/types.go

```diff
@@ -88,6 +88,27 @@ type ConfigT struct {
 	Sources         []SourceT       `json:"sources"`
 	Libraries       LibrariesT      `json:"libraries"`
 	ConnectionFlags ConnectionFlags `json:"flags"`
+	Settings        SettingsT       `json:"settings"`
+}
+
+type SettingsT struct {
+	DataRetention DataRetentionT `json:"dataRetention"`
+}
+
+type DataRetentionT struct {
+	UseSelfStorage     bool                `json:"useSelfStorage"`
+	StorageBucket      StorageBucketT      `json:"storageBucket"`
+	StoragePreferences StoragePreferencesT `json:"storagePreferences"`
```

Comment:
[optional] the JSON tags here (and in the new structs you created) are redundant: https://go.dev/play/p/F9SAkkWcL2j

If you get `useSelfStorage` or even `useSELFStorage` it doesn't matter, it will be parsed. However you will need a JSON tag if the field in the JSON has more characters like `use_self_storage`.